plugin/helpers_config.go

```diff
@@ -28,5 +30,5 @@ func (p *HelpersImpl) CheckRequiredServerConfiguration(req *model.Config) (bool,
 		return false, nil
 	}
 
-	return true, nil
+	return reflect.DeepEqual(&mergedCfg, req), nil
```

Comment:
@isacikgoz, this seems counterintuitive -- `mergedCfg.ToJson() != cfg.ToJson()` should have exactly the desired semantics.

Actually, I wonder if the issue is with the order of parameters passed to `utils.Merge` -- right now, we start with the requested config, and apply the current config, but I think we want the opposite: start with the current config, and apply the requested config. If I make that change, the new test passes without the `reflect.DeepEqual` check here.

However, this crucially assumes that the incoming requested config leaves `nil` all parameters that it doesn't care about. Can you expand the unit tests to define a current configuration for other values in the same/different configuration structs and verify they remain unchanged? I'm wondering too if we can define a test that requires more than just one setting as well.

Reply:
@lieut-data, you are right. If we change the order there is no need to check `reflect.DeepEqual`. I will commit that change. In addition to that I expanded the the test cases to cover the issues you've pointed.